http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/ProxyService.scala

```diff
@@ -169,56 +165,47 @@ class ProxyService(
    */
   def proxyRequest(userInfo: UserInfo, googleProject: GoogleProject, clusterName: ClusterName, request: HttpRequest)(
     implicit ev: ApplicativeAsk[IO, TraceId]
-  ): Future[HttpResponse] =
+  ): IO[HttpResponse] =
     authCheck(userInfo, googleProject, clusterName, ConnectToCluster).flatMap { _ =>
       proxyInternal(userInfo, googleProject, clusterName, request)
     }
 
+  def getTargetHost(googleProject: GoogleProject, clusterName: ClusterName): IO[HostStatus] =
+    Proxy.getTargetHost[IO](clusterDnsCache, googleProject, clusterName)
+
   private def proxyInternal(userInfo: UserInfo,
                             googleProject: GoogleProject,
                             clusterName: ClusterName,
-                            request: HttpRequest): Future[HttpResponse] = {
-    logger.debug(s"Received proxy request for user $userInfo")
+                            request: HttpRequest): IO[HttpResponse] = {
+    logger.info(s"Received proxy request for user $userInfo: ${clusterDnsCache.stats} / ${clusterDnsCache.size}")
     getTargetHost(googleProject, clusterName) flatMap {
       case HostReady(targetHost) =>
         clusterDateAccessedActor ! UpdateDateAccessed(clusterName, googleProject, Instant.now())
         // If this is a WebSocket request (e.g. wss://leo:8080/...) then akka-http injects a
         // virtual UpgradeToWebSocket header which contains facilities to handle the WebSocket data.
         // The presence of this header distinguishes WebSocket from http requests.
-        val responseFuture = request.header[UpgradeToWebSocket] match {
-          case Some(upgrade) => handleWebSocketRequest(targetHost, request, upgrade)
-          case None          => handleHttpRequest(targetHost, request)
+        def responseFuture: Future[HttpResponse] = request.header[UpgradeToWebSocket] match {
+          case Some(upgrade) =>
+            handleWebSocketRequest(targetHost, request, upgrade)
+          case None          =>
+            handleHttpRequest(targetHost, request)
         }
-        responseFuture recoverWith {
+
+        IO.fromFuture(IO(responseFuture)).flatMap {
+          r =>
+            if (r.status.isFailure())
+              IO(logger.info(s"Error response for proxied request: ${r.status}")).as(r)
```

Comment:
Not sure about the volume of logs here either, but this may be nice to have. We could also log the request path here.